### ISSUE TITLE
"missing required posting authority" on broadcast - using custom wrapper

### ISSUE THREAD

**Issue by caspernikus** [steemit/steem#2202]
Hello folks,
maybe somebody around here could help me at my problem. 

Currently I am writing a Swift Wrapper for the Steem Blockchain and I already came pretty far. I think the serialisation works and the signing should (theoretically) work too. I am using the same library for signing as steem-python and I wrote my own Swift-C-Wrapper to use the same functions in the exact same way steem-python uses them. (Library name: sepc256k1)

As a first example I tried to vote for a specified post with following data send to the steem blockchain(node):
```
["method": "call", "jsonrpc": "2.0", "id": 1, "params": ["network_broadcast_api", "broadcast_transaction_synchronous", [{
    expiration = "2018-03-06T14:31:33";
    extensions =     ();
    operations =     (
                (
            vote,
                        {
                author = transisto;
                permlink = "open-letter-to-ranchorelaxo-and-haejin";
                voter = moonrise;
                weight = 10000;
            }
        )
    );
    "ref_block_num" = 58471;
    "ref_block_prefix" = 1044151974;
    signatures =     (
        SIGNATURE
    );
}]]]
``` 
(Ignore the round brackets, they are treated as [] when the data is converted)

The backend (blockchain) is able to parse the data and creates a Public Key from the Signature, sadly the key is not an existing key and ` message = "3030000 tx_missing_posting_auth: missing required posting authority` is returned. 

I have sended the exact same JSON Data via Steem-JS and it worked, so I copied the created Signature and pasted it (for testing) inside my Swift Wrapper, instead of the one I generate. 
Theoretically this should work, since the data sended is the same and the signature did work on steem-js. But also in this case I am receiving ` message = "3030000 tx_missing_posting_auth: missing required posting authority`.

What could went wrong ? 
- Serialised Data wrong ?
- Signature wrong ? (Eventough in Steem-JS it works)
- Something else ?

And yes the Signature is canonical and I can verify it :)

Thanks in advance :)

**Comment by inertia186** [comment]
Did you serialize the extension?

**Comment by caspernikus** [comment]
@inertia186 Yes, the extension is also serialized. For serialization I followed this tutorial: 
https://steemit.com/steem/@xeroc/steem-transaction-signing-in-a-nutshell

But I do not get the same serialization result as xeroc does. Could that be a problem, same data different serialization (I thought I could be because I am using Swift and not python)

**Comment by inertia186** [comment]
Yeah, maybe your serialization is in a different order than expected.

**Comment by caspernikus** [comment]
@inertia186 
So the serialization should return the same values, even tough I am using a different programming language ?

**Comment by inertia186** [comment]
Yes, serialization should be precisely the same no matter what language.

**Comment by inertia186** [comment]
Check out these tests.  In the test called `test_to_bytes` I try to mirror xeroc’s gist example.

https://github.com/inertia186/radiator/blob/master/test/radiator/transaction_test.rb

**Comment by caspernikus** [comment]
@inertia186 Damn... I knew it...
Thank you 👍  I will come back to you after I changed the serialization :)

**Comment by caspernikus** [comment]
@inertia186 Okay serialization is now exactly the same, I checked the digest and it is also the same. So it must be my signing which causes the error ? 

Are you on Discord so we can talk ? My name is the same as here

**Comment by caspernikus** [comment]
@inertia186 Maybe you could have a look at my signing process ? :)
https://gist.github.com/caspernikus/5171e8a89d9b6ebc906375d4913cd8b1

**Comment by inertia186** [comment]
It looks ok, but what I can't tell is if your `serializeTransaction` is returning the correct string.  Assuming it is, I'm not seeing anything obviously wrong with the actual signing.  Here's how radiator does it:

https://github.com/inertia186/radiator/blob/master/lib/radiator/transaction.rb#L217

I'm on this discord: https://discord.gg/BNyw6Tj

**Comment by caspernikus** [comment]
@inertia186 My serialization: https://gist.github.com/caspernikus/a1591a7adc97e3b2cc1f3e3c38c1e33c
